build.gradle

```diff
@@ -203,6 +202,61 @@ task wrapper(type: Wrapper) { gradleVersion = '1.12' }
 
 import javax.tools.ToolProvider
 
+task javadocTarball(type: Tar) {
+  baseName = "gobblin-javadoc-all"
+  destinationDir = new File(project.buildDir, baseName)
+  compression = Compression.GZIP
+  extension = 'tgz'
+  description = "Generates a tar-ball with all javadocs to ${destinationDir}/${archiveName}"
+}
+
+javadocTarball << {
+  def indexFile = new File(destinationDir, "index.md")
+  def version = rootProject.ext.javadocVersion
+  indexFile << """----
```

Comment:
That's intentional. We are storing a multiline string to a file and indentation will become part of the value which is incorrect.